compiler/noirc_evaluator/src/ssa/opt/defunctionalize.rs

```diff
@@ -945,8 +1008,12 @@ mod tests {
         }
         brillig(inline) fn func_2 f2 {
           b0(v0: Field):
-            v2 = call v0(u128 1) -> u1
-            return v2
+            v3 = call f3(u128 1) -> u1
+            return v3
+        }
+        brillig(inline_always) pure fn apply_dummy f3 {
+          b0(v0: u128):
+            return
```

Comment:
At the moment the dummy function return values do not match the signature of the function being called. This IR can panic if the call still exists during inlining. Keeping the IR well formed was the motivation with adding the function parameters. I think it would be good to have IR truly well formed and remove assumptions between the passes. If we prefer this dummy function approach to https://github.com/noir-lang/noir/pull/8677 or accessing zero-length arrays, I will update this PR to have a well formed dummy function (e.g. approach arguments and return values).